service/storage_service.cc

```diff
@@ -429,13 +429,14 @@ future<> storage_service::topology_state_load() {
                 // Populate the table with the state from the gossiper here since storage_service::on_change()
                 // (which is called each time gossiper state changes) may have skipped it because the tokens
                 // for the node were not in the 'normal' state yet
-                co_await update_peer_info(ip);
+                auto info = get_peer_info_for_update(ip);
                 // And then amend with the info from raft
-                co_await _sys_ks.local().update_tokens(ip, rs.ring.value().tokens);
-                co_await _sys_ks.local().update_peer_info(ip, "data_center", rs.datacenter);
-                co_await _sys_ks.local().update_peer_info(ip, "rack", rs.rack);
-                co_await _sys_ks.local().update_peer_info(ip, "host_id", id.uuid());
-                co_await _sys_ks.local().update_peer_info(ip, "release_version", rs.release_version);
+                info.update("tokens", db::system_keyspace::make_tokens_data_value(rs.ring.value().tokens));
+                info.update("data_center", rs.datacenter);
+                info.update("rack", rs.rack);
+                info.update("host_id", id.uuid());
+                info.update("release_version", rs.release_version);
+                co_await _sys_ks.local().update_peer_info(ip, std::move(info));
```

Comment:
Could be a single fixed CQL statement (apart from merging the table name).

Reply:
true
We can pass CQL to uodate_peer_info rather than a column/value map.

Reply:
Passing CQL strings is also not the cleanest.